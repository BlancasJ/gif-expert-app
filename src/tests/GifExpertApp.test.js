import React from 'react';
import GifExpertApp from '../GifExpertApp.js';
import { shallow } from "enzyme";

describe('Tests in GifExpertApp', () => {
  test('should show snapshot of GifExpertApp', () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should show a list of categories', () => {
    const categories = ['One Punch', 'Rick and Morty'];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect( wrapper.find('GifGrid').length ).toBe( categories.length );
  });
});