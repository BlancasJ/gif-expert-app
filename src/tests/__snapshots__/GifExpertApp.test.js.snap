// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in GifExpertApp should show a list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Rick and Morty"
      key="Rick and Morty"
    />
  </ol>
</Fragment>
`;

exports[`Tests in GifExpertApp should show snapshot of GifExpertApp 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
